### ISSUE TITLE
The Plugin only reads 1024 bytes from the pipes to the program started by the plugin

### ISSUE THREAD

**Issue by dinkumoil** [npp-plugins/pork2sausage#10]
The Plugin only reads 1024 bytes from the listener pipes to the program started by the the plugin.

I guess this is due to [this](https://github.com/npp-plugins/pork2sausage/blob/a14648bf449f3ce37fa1b703acdea217d1e5c045/src/Process.cpp#L238) codeblock in the routine to read from the pipe to _StdOut_ of the started program. There is a similar code block [here](https://github.com/npp-plugins/pork2sausage/blob/a14648bf449f3ce37fa1b703acdea217d1e5c045/src/Process.cpp#L293) for reading the pipe to _StdErr_ of the started program.

**Comment by donho** [comment]
@dinkumoil 
What is the bug exactly?
Could you provide the instructions for reproducing it?

**Comment by dinkumoil** [comment]
# Description
If the external program started by the plugin (i.e. the pork-to-sausage converter) produces output that is longer than 1024 bytes, every output beyond this limit is not captured by the plugin.

# Steps to reproduce
1. Install plugin _PorkToSausage_.
2. Navigate to plugin's menu entry and click on submenu entry _Edit User Commands_.
3. Insert the following code and adapt the path to Notepad++ folder to your local requirements.
```
[CaptureOnly1024Bytes]
progPath=C:\Windows\System32\cmd.exe
progCmd=cmd.exe /c "type "<Path-To-Notepad++-Folder>\change.log""
workDir=C:\Windows\System32
replaceSelection=true
```
4. Save document and restart Notepad++.
5. Load file _change.log_ from Notepad++ folder.
6. Press _Ctrl+A_ to select all text of the document.
7. Navigate to plugin's menu entry and click on submenu entry _CaptureOnly1024Bytes_.

# Expected Result
The document's content should be the same as before since it is overwritten with itself.

# Actual Result
The document's content is shortened to the amount of text that correlates to a length of 1024 bytes.

**Comment by dinkumoil** [comment]
My PR fixes this bug by changing the code of the output routine for captured output of external program. It also changes the code of the error output routine, which has the same bug.

Moreover, the PR fixes a second bug of the error output routine - this code is not Unicode aware and leads to garbeled output of error messages, i.e. the output looks like chinese glyphs.

Meanwhile I know you don't like it when PRs mix up fixes for more than one bug. But do not ask me to change that, I uninstalled _Visual Studio_ a long time ago (I'm a _Delphi_ developer) and don't want to reinstall it only to change the commits of my PR.

So, If you do not like my PR simply refuse it. I have already compiled a working version of the plugin by my own, that's enough for me.
